### PR TITLE
Add `qml.workflow.construct_tape(qnode, level)(*args, **kwargs)`

### PR DESCRIPTION
**Pull request by andrijapau**
**Context:**

In an effort to clean-up the code in `QNode`, we need a way to construct tapes from qnodes. This could both be used by the user and internally in the class.

**Description of the Change:**

Create a thin wrapper around the existing `qml.workflow.construct_batch` that returns a single tape. An error will be raised if the `level` argument specified corresponds to more than one tape. 

Example

```python
@qml.qnode(qml.device("default.qubit", shots=10))
def circuit(x):
    qml.RandomLayers(qml.numpy.array([[1.0, 2.0]]), wires=(0,1))
    qml.RX(x, wires=0)
    qml.RX(-x, wires=0)
    qml.SWAP((0,1))
    qml.X(0)
    qml.X(0)
    return qml.expval(qml.X(0) + qml.Y(0))

>>> tape = qml.workflow.construct_tape(circuit)(0.5)
>>> tape.circuit
[RandomLayers(tensor([[1., 2.]], requires_grad=True), wires=[0, 1]),
 RX(0.5, wires=[0]),
 RX(-0.5, wires=[0]),
```

**Benefits:** Code clean-up - allows us to phase out the mutable `QNode.tape` and `QNode.qtape` properties!

**Possible Drawbacks:** None

[sc-72155]

### CODE REVIEW

File: pennylane/workflow/construct_tape.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    expval(X(0) + Y(0))]
+
+    """
+
+    def wrapper(*args, **kwargs):
+
+        batch, _ = qml.workflow.construct_batch(qnode, level)(*args, **kwargs)
+
+        if len(batch) > 1:
+            raise ValueError("Level requested corresponds to more than one tape.")
```
Comment (albi3ro) [inline pennylane/workflow/construct_tape.py:74]:
```suggestion
            raise ValueError("Level requested corresponds to more than one tape. Please use construct_batch instead for this level.")
```

File: pennylane/workflow/construct_tape.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    Args:
+        qnode (QNode): the qnode we want to get the tapes and post-processing for.
+        level (None, str, int, slice): Specifies which stage of the QNode's transform program to use for tape construction.
+            - ``None``: Uses the entire transformation pipeline.
+            - ``"top"``: Uses the top-level transformations.
+            - ``"user"``: Uses transformations up to and including user-defined operations.
+            - ``"device"``: Uses only device-level transformations.
+            - ``"gradient"``: Extracts the gradient-level tape.
+            - ``int``: Specifies the number of transforms to include from the start of the transformation program.
+            - ``slice``: Selects a subset of the transformation program.
```
Comment (albi3ro) [inline pennylane/workflow/construct_tape.py:33]:
```suggestion
        level (None, str, int, slice): Specifies which stage of the QNode's transform program to use for tape construction.

            - ``None``: Uses the entire transformation pipeline.
            - ``"top"``: Uses the top-level transformations.
            - ``"user"``: Uses transformations up to and including user-defined operations.
            - ``"device"``: Uses only device-level transformations.
            - ``"gradient"``: Extracts the gradient-level tape.
            - ``int``: Specifies the number of transforms to include from the start of the transformation program.
            - ``slice``: Selects a subset of the transformation program.
```
Maybe?

**Review by albi3ro — APPROVED** [review]
❤️ 🚀 Looks good :)

File: pennylane/workflow/construct_tape.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    """Constructs the tape for a designated stage in the transform program.
+
+    Args:
+        qnode (QNode): the qnode we want to get the tapes and post-processing for.
+        level (None, str, int, slice): Specifies which stage of the QNode's transform program to use for tape construction.
+
+            - ``None``: Uses the entire transformation pipeline.
+            - ``"top"``: Uses the top-level transformations.
+            - ``"user"``: Uses transformations up to and including user-defined operations.
+            - ``"device"``: Uses only device-level transformations.
```
Comment (astralcai) [inline pennylane/workflow/construct_tape.py:31]: Is this description accurate? I believe for `"device"` we are including all transforms all the way down to the device level, which basically means to use the entire transformation pipeline, equivalent to ``None``.

https://github.com/PennyLaneAI/pennylane/blob/9e1e08e2b0ebc3e2f8352c4aa2cffbeb53d34861/pennylane/workflow/construct_batch.py#L140

File: pennylane/workflow/construct_tape.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+def construct_tape(qnode, level="user"):
+    """Constructs the tape for a designated stage in the transform program.
+
+    Args:
+        qnode (QNode): the qnode we want to get the tapes and post-processing for.
+        level (None, str, int, slice): Specifies which stage of the QNode's transform program to use for tape construction.
+
+            - ``None``: Uses the entire transformation pipeline.
+            - ``"top"``: Uses the top-level transformations.
```
Comment (astralcai) [inline pennylane/workflow/construct_tape.py:29]: I believe that `"top"` means not applying any transformations and returning the original tape as defined.

https://github.com/PennyLaneAI/pennylane/blob/9e1e08e2b0ebc3e2f8352c4aa2cffbeb53d34861/pennylane/workflow/construct_batch.py#L148

File: pennylane/workflow/construct_tape.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            - ``"user"``: Uses transformations up to and including user-defined operations.
+            - ``"device"``: Uses only device-level transformations.
+            - ``"gradient"``: Extracts the gradient-level tape.
+            - ``int``: Specifies the number of transforms to include from the start of the transformation program.
+            - ``slice``: Selects a subset of the transformation program.
```
Comment (astralcai) [inline pennylane/workflow/construct_tape.py:34]: This is probably nitpicking but I wouldn't put `int` and `slice` in the same list as the above. This makes it look like `level=int` is a valid argument.

**Review by astralcai — COMMENTED** [review]
LGTM aside from some ambiguities in the doc-string.

**Review by astralcai — APPROVED** [review]
🥳

**Pull request merged**